### ISSUE TITLE
Request for clarification:  Reinforcement Learning components in code

### ISSUE THREAD

**Issue by lanzhengpeng** [Agent-on-the-Fly/AgentFly#34]
## Issue Description

Hi authors, thank you for open-sourcing this excellent work!  I've been reading the paper and going through the code, and I have some questions about the relationship between the theoretical framework and the actual implementation.

---

##  Question

The paper describes Memento as: 
> "**memory-based online reinforcement learning** over a **memory-augmented MDP**"

And mentions components like:
- State-action-reward tuples **(s_T, a_T, r_T)**
- Case-selection **policy** π_θ
- Experience replay with **Read/Write operations**

However, when I examined the code (particularly `memory/train_memory_retriever.py`), I found: 

```python
# Training loop (lines 276-296)
for epoch in range(epochs):
    for batch in train_loader:
        ids1, mask1, ids2, mask2, labels = batch
        logits = model(ids1, mask1, ids2, mask2)
        
        # Standard cross-entropy loss
        loss = criterion(logits, labels)  
        
        loss.backward()
        optimizer.step()
```

This appears to be **supervised binary classification** (query-case relevance prediction) rather than reinforcement learning.  Specifically, I couldn't find: 

- ❌ Bellman equations or value functions (V(s), Q(s,a))
- ❌ Policy gradient computations (∇log π(a|s) × R)
- ❌ Temporal difference (TD) learning
- ❌ Discount factor γ
- ❌ Cumulative reward calculations

Instead, the code uses:
- ✅ `nn.CrossEntropyLoss` (standard supervised loss)
- ✅ `torch.optim.AdamW` (standard optimizer, not PPO/SAC/etc.)
- ✅ Binary labels from `truth_label` field

---

##  Specific Questions

1. **MDP Formulation**: Could you clarify how the MDP components (states, actions, transitions, policy) map to the actual code implementation? The "policy" seems to be a supervised classifier rather than a learned RL policy.

2. **Reinforcement Learning**:  Are there any RL-specific computations (e.g., value estimation, policy gradients) that I missed in the codebase? Or is the RL framework primarily **conceptual** for organizing the design?

3. **Reward Signal**: The `reward` field in `memory. jsonl` appears to be used as a supervised label for binary classification during retriever training. Is this the intended usage, or is there an RL component that uses these rewards differently?

4. **Training Process**: Could you explain the difference between:
   - The **MDP/RL framework** described in the paper (conceptual level)
   - The **supervised learning** implementation in the code (algorithmic level)

---

##  Why This Matters

I'm very interested in applying similar ideas to my own work, but I want to make sure I understand: 
- Whether this is a **novel RL algorithm** or an **engineering framework** using RL terminology for conceptual organization
- How to properly describe the method when citing or extending it

---

##  Clarification Request

If the RL framework is primarily **conceptual** (i.e., a way to frame the problem) while the implementation uses **supervised learning**, could you: 
1. Explicitly state this in the README or documentation?
2. Add a section discussing the mapping between theory and implementation? 

This would help future researchers understand the design decisions and avoid confusion. 

---

## ✅ What I've Checked

- [x] Read the paper thoroughly
- [x] Reviewed `memory/train_memory_retriever.py`
- [x] Checked `memory/parametric_memory.py` 
- [x] Searched for keywords:  `bellman`, `policy_gradient`, `Q_value`, `discount`, `gamma`
- [x] Examined the training loop and loss functions

---

##  Appreciation

Despite this confusion, I think the **engineering design is excellent**: 
- ✅ Clever use of case-based reasoning
- ✅ Avoids expensive LLM fine-tuning
- ✅ Strong benchmark results
- ✅ Clean, reproducible code

I'm just trying to understand the theoretical foundation better. Thank you for your time!  🙏

---

**Environment**: 
- Repository: Agent-on-the-Fly/Memento
- Commit: `42fbbca` (latest main)
- Files reviewed: `memory/train_memory_retriever.py`, `memory/parametric_memory.py`, `client/*_cbr.py`
